The widget type around feature should work in the "Custom element conversion guide" (#9445)
ckeditor/ckeditor5: packages/ckeditor5-engine/docs/framework/guides/deep-dive/custom-element-conversion.md

	.add( dispatcher => dispatcher.on( 'insert:infoBox', dataDowncastConverter ) );
```

> ### The model to view position mapping

Reviewer: Don't we have to work on reverse direction mapping too? Or does it work out of the box? If so, it'd be good to mention that at the end of this section (e.g. in an info-box ... the Umberto's info-box :D)

Author: The view to model works ok because it uses `findMappedViewAncestor()`. Can you add it?

> ### The model to view position mapping

Reviewer: ```suggestion
Due to the fact that the info box's view structure is much more complex than its model structure we need to take care of one additional aspect to make the  converters work &mdash; position mapping.

### The model to view position mapping
```

> After adding the above downcast converters we would end up with the following view structure:

Reviewer: ```suggestion
The downcast converters shown in the previous section will not yet work correctly. This is how a given model would look after being downcasted:
```

> ```

Reviewer: It'd help to show (maybe in a two-column table) what model translates to what view.

Also, what is the `[]` below for?

<div class="info-box info-box-info">
    []<p>
        Foobar
    </p>
    <div class="info-box-title">Info</div>
    <div class="info-box-content"></div>
</div>
```

> This is not what we need. We expect the content of the `<infoBox>` to be inside the `<div class="info-box-content">`. We have the downcast conversion for the `<infoBox>` and its own structure, but we need to specify where its content should land in the view. By the default it would be converted as a direct child of `<div class="info-box">` (as you can see in the above snippet) but it should go into the `<div class="info-box-content">`. To achieve this, we need to register a callback for the {@link module:engine/conversion/mapper~Mapper#event:modelToViewPosition `Mapper#modelToViewPosition` event}, so the positions inside the model `<infoBox>` element would map to the positions inside the `<div class="info-box-content">` view element.

Reviewer: ```suggestion
As you can see, this is not a correct view structure. The content of the model's `<infoBox>` element ended up directly inside the outer `<div>`. We expect the the `<infoBox>`'s content to be inside the `<div class="info-box-content">`.

We defined downcast conversion for `<infoBox>` itself, but we need to specify where its content should land in its view structure. By default, it is converted as direct children of `<div class="info-box">` (as you can see in the above snippet) but it should go into the `<div class="info-box-content">`. To achieve this, we need to register a callback for the {@link module:engine/conversion/mapper~Mapper#event:modelToViewPosition `Mapper#modelToViewPosition` event}, so positions inside the model `<infoBox>` element would map to positions inside the `<div class="info-box-content">` view element.
```

> ```

Reviewer: If you'll convert the above snippet into a table, this one could be converted too. In any case, be consistent.

Also, I don't feel `[]` helps here.

Author: I have no idea how to represent it as a table. I guess we could go with the proposed snippet and tune it later.

<infoBox infoBoxType="Info">    ->    <div class="info-box info-box-info">
                                          <div class="info-box-title">Info</div>
                                          <div class="info-box-content">
    []<paragraph>               ->            []<p>
        Foobar                  ->                Foobar
    </paragraph>                ->            </p>
                                          </div>
</infoBox>                      ->    </div>
```

> The callback that we need:

Reviewer: ```suggestion
Such mapping can be achieved by registering this callback to {@link XYZ} event:
```

			.add( dispatcher => dispatcher.on( 'insert:infoBox', editingDowncastConverter ) );
		editor.conversion.for( 'dataDowncast' )
			.add( dispatcher => dispatcher.on( 'insert:infoBox', dataDowncastConverter ) );

        // Model to view position mapper is needed since the model <infoBox> content needs to end up in the inner
        // <div class="info-box-content">.
        editor.editing.mapper.on( 'modelToViewPosition', createModelToViewPositionMapper( editor.editing.view ) );

>         editor.data.mapper.on( 'modelToViewPosition', createModelToViewPositionMapper( editor.editing.view ) );

Reviewer: Passing `editor.editing.view`? I didn't check how it's used there but it's smelly.

Author: Yeah, I was talking with you about it some time ago (while refactoring the to-do lists conversion). It's used there only to be able to create a view range.

Reviewer: Do we have a ticket for that? If yes, could you reference it from #7316? If not, please open it.

Author: #9466
